statistics/handle/dump.go

```diff
@@ -82,13 +83,14 @@ func (h *Handle) DumpStatsToJSON(dbName string, tableInfo *model.TableInfo) (*JS
 	return jsonTbl, nil
 }
 
-func (h *Handle) tableStatsToJSON(dbName string, tableInfo *model.TableInfo, physicalID int64) (*JSONTable, error) {
-	tbl, err := h.tableStatsFromStorage(tableInfo, physicalID, true)
-	if err != nil {
-		return nil, errors.Trace(err)
+func (h *Handle) tableStatsToJSON(dbName string, tableInfo *model.TableInfo, physicalID int64, historyStatsExec sqlexec.RestrictedSQLExecutor) (*JSONTable, error) {
+	tbl, err := h.tableStatsFromStorage(tableInfo, physicalID, true, historyStatsExec)
+	if err != nil || tbl == nil {
```

Comment:
I think we should return a specific error message for `err != nil` but `tbl == nil`

Reply:
We do not need to return an error for this case.
"null" will be returned when `tbl == nil`.